src/services/link/hooks/index.js

```diff
@@ -1,14 +1,37 @@
 const hooks = require('feathers-hooks-common');
 const { authenticate } = require('@feathersjs/authentication');
 const globalHooks = require('../../../hooks');
+const logger = require('../../../logger');
+
+const preventDuplicates = async (hook) => {
+	const linkData = hook.data;
+	// get all links to the same target
+	const currentLinks = await hook.app.service('link').find({ query: { target: linkData.target } });
+
+	// check if link to specified target already exists or new link is forced
+	if (currentLinks && currentLinks.total > 0 && !linkData.forceNew) {
+		// if so, set createdAt date to now so that link expires a month from now
+		const id = currentLinks.data[0]._id;
```

Comment:
why asking for multiple links before and handle only the first result here?

Reply:
due to the current nature of the databases, it is likely that there already are multiple links with the same target in the database. For that reason, we only want to use the first result and return it. Same is done in the teams service here: https://github.com/schul-cloud/schulcloud-server/blob/b0fe4ba43c07f3c232977a226d235fbccc7886e1/src/services/teams/index.js#L133